### PR TITLE
Upgrade strimzi 2

### PR DESCRIPTION
**Pull request by tombentley**
### Type of change

_Select the type of your PR_

- Enhancement / new feature

### Description

Adds annotation for explicitly triggering cert renewal.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: operator-common/src/main/java/io/strimzi/operator/cluster/model/Ca.java [hunk 1]
```diff
@@ -76,6 +76,7 @@
     public static final String CA_KEY = "ca.key";
     public static final String CA_CRT = "ca.crt";
     public static final String IO_STRIMZI = "io.strimzi";
+    public static final String ANNO_STRIMZI_IO_FORCE_RENEW = "strimzi.io/force-renew";
```
Comment (ppatierno) [inline operator-common/src/main/java/io/strimzi/operator/cluster/model/Ca.java]: Not for this PR but we should try to centralise these annotations (we have for pod deletion, rolling updates)

File: operator-common/src/main/java/io/strimzi/operator/cluster/model/Ca.java [hunk 2]
```diff
@@ -306,6 +299,34 @@ public void createOrRenew(String namespace, String clusterName, Map<String, Stri
         caKeySecret = secretCertProvider.createSecret(namespace, caKeySecretName, keyData, labels, ownerRef);
     }
 
+    private boolean shouldRenew(X509Certificate currentCert) {
+        String reason = null;
+        boolean result = false;
+        if (forceRenewal) {
```
Comment (ppatierno) [inline operator-common/src/main/java/io/strimzi/operator/cluster/model/Ca.java]: Where the forceRenewal flag is set from the annotation?

File: operator-common/src/main/java/io/strimzi/operator/cluster/model/Ca.java [hunk 3]
```diff
@@ -306,6 +299,34 @@ public void createOrRenew(String namespace, String clusterName, Map<String, Stri
...
+            result = true;
+        } else if (caCertSecret == null
+                || caCertSecret.getData().get(CA_CRT) == null) {
+            reason = "CA certificate secret " + caCertSecretName + " is missing or lacking the key " + CA_CRT;
+            result = true;
+        } else if (caKeySecret == null
+                || caKeySecret.getData().get(CA_KEY) == null) {
+            reason = "CA key secret " + caKeySecretName + " is missing or lacking the key " + CA_KEY;
+            result = true;
+        } else if (caCertSecret != null
```
Comment (stanlyDoge) [inline operator-common/src/main/java/io/strimzi/operator/cluster/model/Ca.java]: We are getting `RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE` here. The nullcheck is already done at line 308.

**Comment by tombentley** [comment]
This is now a PR againsted a merged branch. I'll close this and reopen against `master`.

**Pull request closed without merging**